File: examples/java/Network.java
import com.grafana.foundation.timeseries.PanelBuilder;

public class Network {

    public static PanelBuilder networkReceivedTimeseries() {
        return Common.defaultTimeSeries().
                Title("Network Received").
                Description("Network received (bits/s)").
                Min(0.0).
                Unit("bps").
                FillOpacity(0.0).
                WithTarget(
                        Common.basicPrometheusQuery("rate(node_network_receive_bytes_total{job=\"integrations/raspberrypi-node\", instance=\"$instance\", device!=\"lo\"}[$__rate_interval]) * 8", "{{ device }}")
                );
    }
    public static PanelBuilder networkTransmittedTimeseries() {
        return Common.defaultTimeSeries().
                Title("Network Transmitted").
                Description("Network transmitted (bits/s)").
                Min(0.0).
                Unit("bps").
                FillOpacity(0.0).
                WithTarget(
                        Common.basicPrometheusQuery("rate(node_network_transmit_bytes_total{job=\"integrations/raspberrypi-node\", instance=\"$instance\", device!=\"lo\"}[$__rate_interval]) * 8", "{{ device }}")
                );
    }
}

File: examples/typescript/network.ts
import {PanelBuilder as TimeseriesPanelBuilder} from "../../generated/typescript/src/timeseries";
import {basicPrometheusQuery, defaultTimeseries} from "./common";

export const networkReceivedTimeseries = (): TimeseriesPanelBuilder => {
    return defaultTimeseries()
        .title("Network Received")
        .description("Network received (bits/s)")
        .min(0)
        .unit("bps")
        .fillOpacity(0)
        .withTarget(
            basicPrometheusQuery(`rate(node_network_receive_bytes_total{job="integrations/raspberrypi-node", instance="$instance", device!="lo"}[$__rate_interval]) * 8`, "{{ device }}"),
        );
};

export const networkTransmittedTimeseries = (): TimeseriesPanelBuilder => {
    return defaultTimeseries()
        .title("Network Transmitted")
        .description("Network transmitted (bits/s)")
        .min(0)
        .unit("bps")
        .fillOpacity(0)
        .withTarget(
            basicPrometheusQuery(`rate(node_network_transmit_bytes_total{job="integrations/raspberrypi-node", instance="$instance", device!="lo"}[$__rate_interval]) * 8`, "{{ device }}"),
        );
};
